### PR TITLE
CI updates (#62)

### PR DESCRIPTION
**Pull request by siy**
* Add necessary parts for the Ledger reusable CI workflow
* Add Ragger tests
* Clean up unused components
* Bump SDK and tools version
* Switch to Ledger app builder image and tools
* Bump app version

# Checklist

- [x] App update process has been followed 
- [x] Target branch is `develop` 
- [x] Application version has been bumped

### CODE REVIEW

**Comment by agrojean-ledger** [comment]
@siy this looks good, however, it would be better for your tests to use the Ragger API that provide snapshot comparison, so that you can make sure what is displayed when executing the tests match what is expected. Cf. `navigate_and_compare` or `navigate_until_text_and_compare` navigation APIs from Ragger.

**Comment by siy** [comment]
> @siy this looks good, however, it would be better for your tests to use the Ragger API that provide snapshot comparison, so that you can make sure what is displayed when executing the tests match what is expected. Cf. `navigate_and_compare` or `navigate_until_text_and_compare` navigation APIs from Ragger.

As discussed, I'll prepare separate PR with updated tests.

File: src/main.rs [hunk 1]
```diff
@@ -203,3 +212,21 @@ extern "C" fn sample_main() {
         }
     }
 }
+
+fn display_pending_review(comm: &mut Comm) {
```
Comment (agrojean-ledger) [inline src/main.rs:216]: Why don't you use the built-in pending review function we have added to the SDK ?

https://github.com/LedgerHQ/app-boilerplate-rust/blob/45b68d5f28c071243b9e34d61937aa301663d89a/src/main.rs#L41

https://github.com/LedgerHQ/ledger-device-rust-sdk/blob/377555286d1b94f86ef1e304bb82eb97e453911f/ledger_device_sdk/src/ui/gadgets.rs#L87

Comment (agrojean-ledger) [reply src/main.rs:216]: Ok, those are acceptable justifications as we have yet to take some decisions internally regarding Nano S support and stack usage on this device.

Also, it should be ok for your app to return an error during the pending review if communication from the host is initiated.

**Comment by yogh333** [comment]
👋 Hello,

 [Guideline enforcer workflow](https://github.com/LedgerHQ/app-boilerplate-rust/blob/main/.github/workflows/guidelines_enforcer.yml) is missing from your CI. Could you please add it ? Thanks

**Review by yogh333 — APPROVED** [review]
LGTM

**Review by yogh333 — APPROVED** [review]

**Pull request merged**